fs/littlefs: Remove deprecated *_MEM_POOL_* Kconfig options

de-nordic opened the pull request:
The following options, marked to be deprecated in release 2.5, have
been finally removed from Kconfig:
 FS_LITTLEFS_FC_MEM_POOL_MIN_SIZE
 FS_LITTLEFS_FC_MEM_POOL_MAX_SIZE
 FS_LITTLEFS_FC_MEM_POOL_NUM_BLOCKS

Signed-off-by: Dominik Ermel <dominik.ermel@nordicsemi.no>

de-nordic committed:
fs/littlefs: Remove deprecated *_MEM_POOL_* Kconfig options

The following options, marked to be deprecated in release 2.5, have
been finally removed from Kconfig:
 FS_LITTLEFS_FC_MEM_POOL_MIN_SIZE
 FS_LITTLEFS_FC_MEM_POOL_MAX_SIZE
 FS_LITTLEFS_FC_MEM_POOL_NUM_BLOCKS

Signed-off-by: Dominik Ermel <dominik.ermel@nordicsemi.no>

nashif commented:
dont you have to remove the usage of those Kconfigs in subsys/fs/littlefs_fs.c?

galak commented:
Looks like we need to update subsys/fs/littlefs_fs.c to remove references to CONFIG_FS_LITTLEFS_FC_MEM_POOL_MAX_SIZE and CONFIG_FS_LITTLEFS_FC_MEM_POOL_NUM_BLOCKS

de-nordic commented:
> 
> 
> Looks like we need to update subsys/fs/littlefs_fs.c to remove references to CONFIG_FS_LITTLEFS_FC_MEM_POOL_MAX_SIZE and CONFIG_FS_LITTLEFS_FC_MEM_POOL_NUM_BLOCKS

Yeah, I did that but managed to mess up and loose the commit while moving stuff. Give me some time to sulk and I will fix this :(

de-nordic committed:
fs/littlefs: Completely switch from mem pool to heap

The LittleFS file cache has been completely switched from k_mem_slab
to heap allocated.  The *_MEM_POOL_* kconfig variables no longer
affect the code.

Signed-off-by: Dominik Ermel <dominik.ermel@nordicsemi.no>

de-nordic commented:
Not a good day...

de-nordic commented:
@galak Seems to be a problem with past release notes:
```

-- Configuring done
-- Generating done
-- Build files have been written to: /home/runner/work/zephyr/zephyr/doc/_build
cmake --build _build -- htmldocs  # -v # VERBOSE=1
[0/3] Running gen_devicetree_rest.py /home/runner/work/zephyr/zephyr/doc/_build/src/reference/devicetree
[2/3] Running gen_kconfig_rest.py /home/runner/work/zephyr/zephyr/doc/_build/src/reference/kconfig/
[2/3] Generating HTML documentation with /home/runner/.local/bin/sphinx-build -q;-W;-j;auto

Warning, treated as error:
/home/runner/work/zephyr/zephyr/doc/_build/src/releases/release-notes-2.5.rst:826:unknown option: CONFIG_FS_LITTLEFS_FC_MEM_POOL
Error: The operation was canceled.
```

galak commented:
@nashif @mbolivar-nordic do you know how we dealt with this in the past?

nashif commented:
> @nashif @mbolivar-nordic do you know how we dealt with this in the past?

do not use :option: in release notes, just add a double quote for emphasis

mbolivar-nordic commented:
For an example of what @nashif is saying: https://github.com/zephyrproject-rtos/zephyr/commit/8165008f44beb68f686ac37b6db99f94a6ced71c#diff-2fa87458d679dcd39c7f54732b4b885ff5261ebaf4469919cc29133573a69bb5L136

de-nordic committed:
release: Fix LittleFS entry in 2.5 release notes

The document build system has been complaining about use of
no longer existing Kconfig option.

Signed-off-by: Dominik Ermel <dominik.ermel@nordicsemi.no>

galak commented on subsys/fs/littlefs_fs.c:
```diff
@@ -37,79 +37,35 @@ static K_MEM_SLAB_DEFINE(file_data_pool, sizeof(struct lfs_file_data),
- * mem-pool configuration options.
- */
 #if (CONFIG_FS_LITTLEFS_FC_HEAP_SIZE - 0) <= 0
+/* Autho-generate heap size from cache size and number of files */
```
was this suppose to be auto-generate?

de-nordic replied:
Yes. The Kconfig comment for the `CONFIG_FS_LITTLEFS_FC_HEAP_SIZE` states that if <=0 it will be picked for user.

galak replied:
I guess the question was is this a typo? `Autho` v `Auto`

de-nordic replied:
Yep. You got me there. Fixing.

galak commented:
LGTM.  Just one question about a minor typo
